lib/git.ml

```diff
@@ -0,0 +1,4 @@
+let is_git_uri uri =
```

Comment:
Git uri are not terminated by `.git`. They can take a lot of forms and we should not try to parse them. I suggest to move this function closer to where it is used because it does not make sense on its own and is only used to make guess.

Reply:
Maybe we could rename it `has_git_extension`, in which case it describes exactly what happens and I think it's then fairly harmless to leave it here.

Reply:
I don't like keeping it here because it adds a new module, new test files and the presence of the ".git" extension does not mean anything to git.

Reply:
Why would adding new modules or tests be a liablility?

It might not mean anything to git but it still is widely used, even https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols puts one in every single examples.

My take on this is having a module for git related helper functions make sense no matter what. This is just a step in a refactoring and with time more functions will most certainly end up here but even they don't that doesn't mean we most not create modules to separate concerns. This makes it easier when working on the `duniverse` repo to look for them if they seat in the right place.

Reply:
To sum up our offline discussion: we'll move this function to a `Uri_utils` as a `Git` module would be better suited for functions that apply in a git context and this is used to determine whether we are in a git context or not.